### PR TITLE
Fix to lean 4 documentation for calculational proofs

### PR DESCRIPTION
**Pull request by Shiney**
Fix the documentation about calculational proofs to work with Lean 4

### CODE REVIEW

File: templates/extras/calc.md [hunk 1]
```diff
@@ -52,31 +30,20 @@ proof is complete. Use `_` as in the example above or `sorry` to stand
...
-have H : A = D,
-{ calc A = B : sorry
-  ...    = C : sorry
-  ...    = D : _,
-  sorry },
-```
-and the `_` can be left in as they generate a subgoal to be resolved
-after calc (here by the last `sorry`).
-
 (Idle thought: could one write a VS Code snippet to write this skeleton?)
```
Comment (PatrickMassot) [inline templates/extras/calc.md:47]: This comment is outdated. There is now a code action doing it. More generally this documentation should now explain the calc widget.

File: templates/extras/calc.md [hunk 2]
```diff
@@ -86,31 +53,37 @@ the operators, but actually `calc` will work with any relation for which
 the corresponding transitivity statement is tagged `[trans]`:
```
Comment (PatrickMassot) [inline templates/extras/calc.md:53]: This is outdated. This attribute no longer exists.

File: templates/extras/calc.md (hunk 1 above)
Comment (Shiney) [reply templates/extras/calc.md:47]: Do you have details of the calc widget? I can't work it out

File: templates/extras/calc.md (hunk 2 above)
Comment (Shiney) [reply templates/extras/calc.md:53]: Whoops sorry missed that one

File: templates/extras/calc.md (hunk 1 above)
Comment (Shiney) [reply templates/extras/calc.md:47]: Didn't know about this having a look into this

Comment (Shiney) [reply templates/extras/calc.md:47]: Sorry can't work out how to use this

File: templates/extras/calc.md [hunk 3]
```diff
@@ -52,65 +30,60 @@ proof is complete. Use `_` as in the example above or `sorry` to stand
...
 
-In tactic mode calc should be terminated with a comma:
-```lean
-have H : A = D,
-{ calc A = B : sorry
-  ...    = C : sorry
-  ...    = D : _,
-  sorry },
-```
-and the `_` can be left in as they generate a subgoal to be resolved
```
Comment (eric-wieser) [inline templates/extras/calc.md:77]: This sentence is still true (or at least, it's true with `?_` instead of `_`.

Comment (Shiney) [reply templates/extras/calc.md:77]: Thanks will try this out soon.

File: templates/extras/calc.md (hunk 1 above)
Comment (Shiney) [reply templates/extras/calc.md:47]: Apparently you've made a video about this I'll go find it

File: templates/extras/calc.md (hunk 3 above)
Comment (Shiney) [reply templates/extras/calc.md:77]: Okay added that information back in

File: templates/extras/calc.md (hunk 1 above)
Comment (Shiney) [reply templates/extras/calc.md]: added a link to a video of you demonstrating this.

File: templates/extras/calc.md [hunk 4]
```diff
@@ -1,49 +1,27 @@
...
 ```lean
 example (a b c : ℕ) (H1 : a = b + 1) (H2 : b = c) : a = c + 1 :=
-calc a = b + 1 : H1
-...    = c + 1 : by rw H2
+calc a = b + 1 := H1
+     _ = c + 1 := by rw[H2]
 ```
 
-`calc` is also available in tactic mode. You can leave `_`s to create a
+`calc` is also available in tactic mode.
```
Comment (YaelDillies) [inline templates/extras/calc.md:16]:
```suggestion
`calc` is also available in tactic mode. You can leave `?_`s to create a
```

File: templates/extras/calc.md [hunk 5]
```diff
@@ -1,49 +1,27 @@
...
+[section 4](https://lean-lang.org/theorem_proving_in_lean4/quantifiers_and_equality.html#calculational-proofs).
 
 Basic example usage:
 
 ```lean
 example (a b c : ℕ) (H1 : a = b + 1) (H2 : b = c) : a = c + 1 :=
-calc a = b + 1 : H1
-...    = c + 1 : by rw H2
+calc a = b + 1 := H1
+     _ = c + 1 := by rw[H2]
```
Comment (YaelDillies) [inline templates/extras/calc.md:13]: This style is preferred:
```suggestion
     _ = c + 1 := by rw [H2]
```

Comment (YaelDillies) [inline templates/extras/calc.md:13]: Can you adapt to putting the `calc` on its own line here and everywhere else? This is preferred style now
```suggestion
calc
  a = b + 1 := H1
  _ = c + 1 := by rw[H2]
```

File: templates/extras/calc.md [hunk 6]
```diff
@@ -1,49 +1,27 @@
...
 ```lean
-example (a b c : ℕ) (H1 : a = b + 1) (H2 : b = c) : a = c + 1 :=
-begin
-  calc a = b + 1 : H1
-  ...    = c + 1 : _,
-  { rw H2 }
-end
+example (a b c : ℕ) (H1 : a = b + 1) (H2 : b = c) : a = c + 1 := by
+  calc a = b + 1 := H1
+       _ = c + 1 := by rw[H2]
```
Comment (YaelDillies) [inline templates/extras/calc.md:21]:
```suggestion
example (a b c : ℕ) (H1 : a = b + 1) (H2 : b = c) : a = c + 1 := by
  calc
    a = b + 1 := ?_
    _ = c + 1 := ?_
  · exact H1
  · rw [H2]
```

File: templates/extras/calc.md [hunk 7]
```diff
@@ -52,65 +30,70 @@ proof is complete. Use `_` as in the example above or `sorry` to stand
 for missing justifications. `sorry` will supress error messages
 entirely, while `_` will generate a guiding error message.
 
-If the structure of calc is incorrect (e.g., missing `:` or the
+If the structure of calc is incorrect (e.g., missing `:=` or the
 justification after it), you may see error messages that are obscure
-and/or red squiggles that end up under a random `...`. To avoid these,
+and/or red squiggles that end up under a random `_`. To avoid these,
 you might first populate a skeleton proof such as:
```
Comment (YaelDillies) [inline templates/extras/calc.md:36]: I'm not actually sure this is true anymore. You should test precisely what happens in several common scenarios (missing `_` after the `:=`, missing `:=`, missing `_` before the relation...) and report back.

File: templates/extras/calc.md [hunk 8]
```diff
@@ -52,65 +30,70 @@ proof is complete. Use `_` as in the example above or `sorry` to stand
...
-...    = D : sorry
+calc A = B := sorry
+     _ = C := _
+     _ = D := sorry
 ```
 
 and then fill in the `sorry` and `_` gradually.
 
-In tactic mode calc should be terminated with a comma:
+In tactic mode a `?_` can be used to generate a subgoal to be resolved.
```
Comment (YaelDillies) [inline templates/extras/calc.md:47]: That's already mentioned above

File: templates/extras/calc.md [hunk 9]
```diff
@@ -52,65 +30,70 @@ proof is complete. Use `_` as in the example above or `sorry` to stand
...
+     calc A = B := sorry
+          _ = C := ?_
+          _ = D := sorry
+     sorry
 ```
-and the `_` can be left in as they generate a subgoal to be resolved
-after calc (here by the last `sorry`).
 
-(Idle thought: could one write a VS Code snippet to write this skeleton?)
+There is also a Calc widget that makes this easier here is a [video](https://youtu.be/8MFGhOWeCNE?t=1834). demonstrating it.
```
Comment (YaelDillies) [inline templates/extras/calc.md:57]:
```suggestion
There is also a Calc widget that makes this easier here is a [video](https://youtu.be/8MFGhOWeCNE?t=1834) demonstrating it.
```

File: templates/extras/calc.md [hunk 10]
```diff
@@ -52,65 +30,70 @@ proof is complete. Use `_` as in the example above or `sorry` to stand
...
+    _ < d     := h3
+
  ```
 
 What is actually going on here? The proofs themselves are not a mystery,
-for example `nat.succ_le_succ h2` is a proof of `b + 1  ≤ c + 1`. The
+for example `Nat.succ_le_succ h2` is a proof of `b + 1  ≤ c + 1`. The
 clever part is that lean can put all of these together to correctly
 deduce that if `U = V < W ≤ X < Y` then `U < Y`. The way this is done,
 Kevin thinks (can someone verify this?) is that Lean continually tries
```
Comment (YaelDillies) [inline templates/extras/calc.md:99]: Can you verify this?

File: templates/extras/calc.md [hunk 11]
```diff
@@ -132,37 +115,33 @@ doing this? The easiest case is when one of `op1` and `op2`
...
-```
-
-and it is easily seen that these lemmas can be used to justify the example in the manual.
+operator is the equality operator, Lean looks through the instances of
+`Trans` and applies these instead.
 
 ## Using user-defined operators
 
-It is as simple as tagging the relevant results with `trans`. For example
+It is as simple as tagging the relevant adding new `Trans`. For example
```
Comment (YaelDillies) [inline templates/extras/calc.md:127]: This sentence is grammatically incorrect

File: templates/extras/calc.md (hunk 8 above)
Comment (PatrickMassot) [inline templates/extras/calc.md:47]:
```suggestion
```

File: templates/extras/calc.md (hunk 11 above)
Comment (PatrickMassot) [inline templates/extras/calc.md:127]:
```suggestion
It is as simple as creating the relevant `Trans` instances. For example
```

**Pull request merged**